### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by joshcooper** [comment]
nit there's an extra "(squash) - isntance variable fix" commit

**Comment by joshcooper** [comment]
commit https://github.com/puppetlabs/puppetlabs-network_device_core/pull/1/commits/dacb695bb47b8f41091c6ce421ec4d265730aea6 extracts the following files from `lib`:

```
lib/puppet/provider/cisco.rb
lib/puppet/provider/interface/cisco.rb
lib/puppet/provider/vlan/cisco.rb
lib/puppet/type/interface.rb
lib/puppet/type/router.rb
lib/puppet/type/vlan.rb
lib/puppet/util/network_device/cisco.rb
lib/puppet/util/network_device/cisco/device.rb
lib/puppet/util/network_device/cisco/facts.rb
lib/puppet/util/network_device/cisco/interface.rb
lib/puppet/util/network_device/ipcalc.rb
lib/puppet/util/network_device/transport/ssh.rb
lib/puppet/util/network_device/transport/telnet.rb
```

To confirm `lib/puppet/util/network_device.rb` is expected to stay in puppet? I assume it's considered puppet API?

Are there other network device related modules that are relying on `lib/puppet/util/network_device/ipcalc.rb` or the transports `lib/puppet/util/network_device/transport/{ssh,telnet}.rb`?

File: lib/puppet/util/network_device/cisco/device.rb [hunk 1]
```diff
@@ -141,7 +141,7 @@ def parse_interface(name)
       if l =~ %r{Auto Speed \(.+\),} || l =~ %r{Auto Speed ,} || l =~ %r{Auto-speed}
         resource[:speed] = :auto
       end
-      if l =~ /, (.+)Mb\/s/
```
Comment (joshcooper) [inline lib/puppet/util/network_device/cisco/device.rb]: When converting to `%r`, you'll want to remove the escaping of forward slashes:

```
irb(main):004:0> /, (.+)Mb\/s/ == %r{, (.+)Mb\/s}
=> false
irb(main):005:0> /, (.+)Mb\/s/ == %r{, (.+)Mb/s}
=> true
```

**Comment by melissa** [comment]
Yes, @shermdog and I determined that `ipcalc.rb` and `transport/{ssh,telent}.rb` should be removed and `lib/puppet/util/network_device.rb` should stay in puppet.

I'll fix the regex changes now

**Comment by melissa** [comment]
@joshcooper updated!

File: README.md [hunk 2]
```diff
@@ -1 +1,95 @@
-# network_device_core
\ No newline at end of file
+
+# network_device_core
+
+Welcome to your new module. A short overview of the generated parts can be found in the PDK documentation at https://puppet.com/pdk/latest/pdk_generating_modules.html .
+
+The README template below provides a starting point with details about what information to include in your README.
```
Comment (joshcooper) [inline README.md]: readme needs updating and `REFERENCE.md`

**Comment by melissa** [comment]
Okay, README.md updated, and REFERENCE.md added

File: README.md [hunk 3]
```diff
@@ -1,95 +1,87 @@
...
     * [Beginning with network_device_core](#beginning-with-network_device_core)
 3. [Usage - Configuration options and additional functionality](#usage)
-4. [Limitations - OS compatibility, etc.](#limitations)
-5. [Development - Guide for contributing to the module](#development)
+4. [Reference - User documentation](#reference)
+5. [Limitations - OS compatibility, etc.](#limitations)
+6. [Development - Guide for contributing to the module](#development)
 
 ## Description
+The functionality in this module has been deprecated and superseded by the puppetlabs/cisco_ios module. Please use that modules instead.
```
Comment (joshcooper) [inline README.md]: typo `that module`

File: README.md [hunk 4]
```diff
@@ -1,95 +1,87 @@
...
 
 ## Setup
 
-### What network_device_core affects **OPTIONAL**
-
-If it's obvious what your module touches, you can skip this section. For example, folks can probably figure out that your mysql_instance module affects their MySQL instances.
+### Setup Requirements
 
-If there's more that they should know about, though, this is the place to mention:
+In order to use the network_device module, you must have
```
Comment (joshcooper) [inline README.md]: looks like some verbiage is missing?

**Review by joshcooper — APPROVED** [review]

**Pull request merged**